app/src/main/java/com/epfl/drawyourpath/community/TournamentCreationFragment.kt

```diff
@@ -30,6 +49,22 @@ class TournamentCreationFragment : Fragment(R.layout.fragment_tournament_creatio
     // TODO add visibility to tournament
     private lateinit var visibility: RadioGroup
 
+    // data class used when the user has chosen the parameters of the tournament.
+    data class TournamentParameters(
+        val name: String,
+        val description: String,
+        val startDate: LocalDateTime,
+        val endDate: LocalDateTime,
+        val visibility: Tournament.Visibility,
+    )
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is Activity) {
+            mActivity = context
+        }

```

Comment:
Well when I tried to make the toasts appear it didn't work through `activity,` but I must have made something wrong because you're right, it doesn't change anything. Changing it back to `activity`. Good catch!